feat(Deployments): support Job Variables

felixpelletier opened the pull request:
## Summary
Supports setting Job Variables on Deployment resources.

Related to https://github.com/PrefectHQ/terraform-provider-prefect/issues/238

Note: I'm a Go newbie, but I mainly just followed the very similar PR here: https://github.com/PrefectHQ/terraform-provider-prefect/pull/245

## Testing
I ran the unit tests, but I don't have an appropriate setup to run the acceptance tests. 
If someone could help with that, I'd be very grateful 😉

felixpelletier committed:
feat(Deployments): support Job Variables

mitchnielsen commented:
This is great, thanks for your contribution @felixpelletier! Will take a look 👍🏼

mitchnielsen committed:
Use 'env' key for job_variables test and example

The 'env' key is "the only job variable that is configurable for all
work pool types"
(https://docs.prefect.io/latest/guides/deployment/overriding-job-variables/)
so let's use that as the example and as the test to confirm it works as
expected.

mitchnielsen commented:
Hi @felixpelletier, I pushed one small change 309ab47. I found that `env` was a key that works for all work pool types, so I wanted to use that in the example and tests to confirm that your changes work as expected (and they do 👍🏼).

mitchnielsen committed:
Address golangci-lint var naming changes

mitchnielsen reviewed (approved):
This looks great, thanks again @felixpelletier 🎉 

- New field added in correct places
- Tests added and passing
- Docs updated

Will approve and hand to @parkedwards for a final lookover since I ended up pushing a couple small changes.
